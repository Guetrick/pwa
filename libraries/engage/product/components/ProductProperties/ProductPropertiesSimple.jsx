import React from 'react';
import PropTypes from 'prop-types';
import { SurroundPortals } from '@shopgate/pwa-common/components';
import { PRODUCT_PROPERTIES } from '@shopgate/pwa-common-commerce/product/constants/Portals';
import { ProductPropertiesRows } from './ProductPropertiesRows';
import { container } from './ProductProperties.style';

/**
 * @param {Object} props The component props.
 * @returns {JSX}
 */
export function ProductPropertiesSimple({ properties }) {
  return (
    <SurroundPortals portalName={PRODUCT_PROPERTIES}>
      {(properties !== null && properties.length !== 0) && (
        <div className={container}>
          <table>
            <tbody>
              <ProductPropertiesRows properties={properties} />
            </tbody>
          </table>
        </div>
      )}
    </SurroundPortals>
  );
}

ProductPropertiesSimple.propTypes = {
  properties: PropTypes.arrayOf(PropTypes.shape()),
};

ProductPropertiesSimple.defaultProps = {
  properties: null,
};
