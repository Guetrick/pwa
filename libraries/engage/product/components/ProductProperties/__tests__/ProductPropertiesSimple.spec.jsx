import React from 'react';
import { shallow, mount } from 'enzyme';
import { ProductPropertiesSimple } from '../ProductPropertiesSimple';

const properties = [
  {
    label: 'test1',
    value: '123',
  },
  {
    label: 'test2',
    value: '456',
  },
];

describe('<ProductPropertiesSimple />', () => {
  it('should not render if no data is available', () => {
    const wrapper = shallow(<ProductPropertiesSimple properties={null} />);
    expect(wrapper.find('table').length).toEqual(0);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render table with properties', () => {
    const wrapper = mount(<ProductPropertiesSimple properties={properties} />);
    expect(wrapper.find('table').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
