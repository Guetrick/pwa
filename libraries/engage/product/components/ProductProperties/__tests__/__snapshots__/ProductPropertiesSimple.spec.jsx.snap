// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPropertiesSimple /> should not render if no data is available 1`] = `
<SurroundPortals
  portalName="product.properties"
  portalProps={null}
/>
`;

exports[`<ProductPropertiesSimple /> should render table with properties 1`] = `
<ProductPropertiesSimple
  properties={
    Array [
      Object {
        "label": "test1",
        "value": "123",
      },
      Object {
        "label": "test2",
        "value": "456",
      },
    ]
  }
>
  <SurroundPortals
    portalName="product.properties"
    portalProps={null}
  >
    <Portal
      name="product.properties.before"
      props={null}
    />
    <Portal
      name="product.properties"
      props={null}
    >
      <div
        className={
          Object {
            "data-css-1x8awm9": "",
          }
        }
      >
        <table>
          <tbody>
            <ProductPropertiesRows
              properties={
                Array [
                  Object {
                    "label": "test1",
                    "value": "123",
                  },
                  Object {
                    "label": "test2",
                    "value": "456",
                  },
                ]
              }
            >
              <ProductPropertiesRowUngrouped
                properties={
                  Array [
                    Object {
                      "label": "test1",
                      "value": "123",
                    },
                    Object {
                      "label": "test2",
                      "value": "456",
                    },
                  ]
                }
              >
                <ProductPropertiesRow
                  key="test1-123"
                  label="test1"
                  value="123"
                >
                  <tr
                    key="test1123"
                  >
                    <td
                      className={
                        Object {
                          "data-css-fl9ufm": "",
                        }
                      }
                    >
                      test1
                    </td>
                    <td
                      className={
                        Object {
                          "data-css-fl9ufm": "",
                        }
                      }
                      data-test-id="property: 123"
                    >
                      123
                    </td>
                  </tr>
                </ProductPropertiesRow>
                <ProductPropertiesRow
                  key="test2-456"
                  label="test2"
                  value="456"
                >
                  <tr
                    key="test2456"
                  >
                    <td
                      className={
                        Object {
                          "data-css-fl9ufm": "",
                        }
                      }
                    >
                      test2
                    </td>
                    <td
                      className={
                        Object {
                          "data-css-fl9ufm": "",
                        }
                      }
                      data-test-id="property: 456"
                    >
                      456
                    </td>
                  </tr>
                </ProductPropertiesRow>
              </ProductPropertiesRowUngrouped>
              <ProductPropertiesSubGroups
                properties={
                  Array [
                    Object {
                      "label": "test1",
                      "value": "123",
                    },
                    Object {
                      "label": "test2",
                      "value": "456",
                    },
                  ]
                }
              />
            </ProductPropertiesRows>
          </tbody>
        </table>
      </div>
    </Portal>
    <Portal
      name="product.properties.after"
      props={null}
    />
  </SurroundPortals>
</ProductPropertiesSimple>
`;
